### PR TITLE
Chore/metadata refactoring 6

### PR DESCRIPTION
**Pull request by mroz22**
cherry-picks from #9130 

- 659e69881d85c52f0d7d61823d672b4dd0992258 saving to long-term storage is now incorrect - it saves entire metadat object, which is not expected of course.
- 90589577a61a17c1a78c2c222737d29eccddef8d - just some related typo
- ab3659ef18606ecfa468268f2b4da2787a9be54e - cipherKeyValue should add lock. it has dialogue on device so it makes sense to block UI and buttons in this case too.

### CODE REVIEW

File: suite-common/wallet-core/src/discovery/discoveryThunks.ts [hunk 1]
```diff
@@ -134,7 +134,7 @@ const handleProgressThunk = createThunk(
             }),
         );
 
-        // change auth confirm field only if metadata are not enabled
+        // change auth confirm field only if metadata is not enabled
```
Comment (matejkriz) [inline suite-common/wallet-core/src/discovery/discoveryThunks.ts]: It's quite a linguistic conundrum 😆 I believe "are" was more correct here, but of course it doesn't matter here...

Comment (mroz22) [reply suite-common/wallet-core/src/discovery/discoveryThunks.ts]: huh you are right :D

Comment (mroz22) [reply suite-common/wallet-core/src/discovery/discoveryThunks.ts]: removed

**Review by matejkriz — APPROVED** [review]
looks good, makes sense 👌

**Pull request merged**